lib/externals/scms/git_project.rb

```diff
@@ -3,7 +3,16 @@
 module Externals
   class GitProject < Project
     def default_branch
-      'master'
+      branch = begin
+                 `git config --global init.defaultBranch`
+               rescue
+                 ""
```

Comment:
we probably shouldn't swallow this up and probably need to check the exit code for this. I can take a peak at other parts of the code to see how I do this elsewhere and I'll make some suggestions. This is really old code and looks like this function is from over 12 years ago, lol!

Reply:
Yes, you are right. I was being too lazy. Looking at `git config --help`, I think that the correct behavior would be to just pass the error up to the caller. 

These error conditions should be caught in code review. 
       •   The section or key is invalid (ret=1),
       •   no section or name was provided (ret=2),
       •   you try to unset an option which does not exist (ret=5),
       •   you try to unset/set an option for which multiple lines match (ret=5), or
       •   you try to use an invalid regexp (ret=6).
       •   the config file cannot be written (ret=4),

And this error conditions would be a problem in the user's system that we don't control.
       •   the config file is invalid (ret=3),

So no additional error handling is really necessary.

Reply:
Interesting. I think we might be able to get away with simply this then:

```ruby
    def default_branch
      `git config init.defaultBranch`.presence || "master"
    end
```

Though we might also be able to delete this code if it's actually unreachable.